### DISCUSSION TITLE
Automation Query rows test does not match live result.

### DISCUSSION THREAD

**Discussion by JClarke90** [Budibase/budibase#11883] [Help]
Trying to figure out if I've done something wrong. 

I have an automation task that queries rows, when I test this in the autmation configurator it returns the live data from the budubase DB which is what I expect. Images below:

![image](https://github.com/Budibase/budibase/assets/12259768/d00a7258-5aac-4947-9978-e08ddcd30b04)

![image](https://github.com/Budibase/budibase/assets/12259768/4add1141-7832-458d-9276-16f288f9fdd7)

```
{
      "_id": "ro_ta_c4c3be41c309427b8005a5897bee50ba_9844b9168f284e979c4e9efb2fcbbda9",
      "_rev": "15-3f9a085c0994338665724390a1c5c5c8",
      "tableId": "ta_c4c3be41c309427b8005a5897bee50ba",
      "name": "NT0APPOTOT01",
      "id": "***NT0APPOTOT01",
      "type": "row",
      "resourceGroup": "OTOACCESS-TST-001",
      "scheduleActive": true,
      "saturday": false,
      "sunday": false,
      "monday": false,
      "mondayStartTime": "2000-01-01T23:55:00.000Z",
      "mondayStopTime": "2000-01-01T00:30:00.000Z",
      "tuesday": true,
      "tuesdayStartTime": "2000-01-01T01:20:00.000Z",
      "tuesdayStopTime": "2000-01-01T01:40:00.000Z",
      "wednesday": false,
      "wednesdayStartTime": null,
      "wednesdayStopTime": null,
      "thursday": false,
      "thursdayStartTime": null,
      "thursdayStopTime": null,
      "friday": false,
      "fridayStartTime": null,
      "fridayStopTime": null,
      "createdAt": "2023-09-26T01:17:37.045Z",
      "updatedAt": "2023-09-26T01:17:37.045Z"
    }
```

But when it runs in production it isn't using the most up to date data in the budibase DB, the updatedAt time is before the one above despite running after this time. Shown below.

![image](https://github.com/Budibase/budibase/assets/12259768/d2ed8b8f-8079-44fa-9542-fbab41a43d52)

```    
{
      "_id": "ro_ta_c4c3be41c309427b8005a5897bee50ba_9844b9168f284e979c4e9efb2fcbbda9",
      "_rev": "14-fd3bcd3b1e745b6c36a8fb3c80974d98",
      "tableId": "ta_c4c3be41c309427b8005a5897bee50ba",
      "name": "NT0APPOTOT01",
      "id": "***NT0APPOTOT01",
      "type": "row",
      "resourceGroup": "OTOACCESS-TST-001",
      "scheduleActive": true,
      "saturday": false,
      "sunday": false,
      "monday": true,
      "mondayStartTime": "2000-01-01T23:55:00.000Z",
      "mondayStopTime": "2000-01-01T00:30:00.000Z",
      "tuesday": false,
      "tuesdayStartTime": null,
      "tuesdayStopTime": null,
      "wednesday": false,
      "wednesdayStartTime": null,
      "wednesdayStopTime": null,
      "thursday": false,
      "thursdayStartTime": null,
      "thursdayStopTime": null,
      "friday": false,
      "fridayStartTime": null,
      "fridayStopTime": null,
      "createdAt": "2023-09-26T00:29:34.840Z",
      "updatedAt": "2023-09-26T00:29:34.840Z"
    }
```

Am I missing something?

**Comment by andz-bb** [comment]
hey @JClarke90, if you are using Budibase DB as your data source there are two DBs in the background - a development DB that is used in the builder and a production DB used in your published app. If you added the newer row in the builder or preview you will need to publish your app to have those changes sync to the production DB. You can [read more about the dev/prod data sync in our docs.
](https://docs.budibase.com/docs/budibasedb#data-sync-between-environments)

If you have already published your app, it might be worth looking at your sorting and filtering on the Query Rows step. What is your use case here - do you want to always return the latest row in this step?

**Comment by JClarke90** [comment]
Hi @andz-bb thanks for the info, shortly after posting I noticed that when using the published app it was working correctly.

Good to know how the Budibase DB works as it will help me avoid this in the future.

As for the query, no it's not the latest row I need, I'm grabbing specific rows based on the day and time and running them through an API query, it looks like its all working successfully in production now though. 

Although I did have one more query, in the designer you can set the date picker to time only, but in BB DB it still has the date attached, is there any way to have just a time field in the BB DB? I've used JS to work around this in automation but it would be cleaner to have a time only field in the DB.
